examples_tests/35.GeometryCreator/main.cpp

```diff
@@ -221,17 +235,88 @@ int main()
 		std::make_pair(icosphereGeometry, gpuIcosphere)
 	};
 	
-	uint64_t lastFPSTime = 0;
-	while(device->run() && receiver.keepOpen())
+	QToQuitEventReceiver escaper;
+	CommonAPI::InputSystem::ChannelReader<IMouseEventChannel> mouse;
+	CommonAPI::InputSystem::ChannelReader<IKeyboardEventChannel> keyboard;
+
+	core::vectorSIMDf cameraPosition(0, 5, -10);
+	matrix4SIMD projectionMatrix = matrix4SIMD::buildProjectionMatrixPerspectiveFovLH(core::radians(60), float(WIN_W) / WIN_H, 0.001, 1000);
+	Camera camera = Camera(cameraPosition, core::vectorSIMDf(0, 0, 0), projectionMatrix, 10.f, 1.f);
+	auto lastTime = std::chrono::system_clock::now();
+
+	constexpr size_t NBL_FRAMES_TO_AVERAGE = 100ull;
+	size_t frame_count = 0ull;
+	double time_sum = 0;
+	double dtList[NBL_FRAMES_TO_AVERAGE] = {};
+	for (size_t i = 0ull; i < NBL_FRAMES_TO_AVERAGE; ++i)
+		dtList[i] = 0.0;
+
+	nbl::core::smart_refctd_ptr<nbl::video::IGPUSemaphore> render_finished_sem;
+	while(escaper.keepOpen())
 	{
-		driver->beginScene(true, true, video::SColor(255,255,255,255) );
-
-        //! This animates (moves) the camera and sets the transforms
-		camera->OnAnimate(std::chrono::duration_cast<std::chrono::milliseconds>(device->getTimer()->getTime()).count());
-		camera->render();
+		auto renderStart = std::chrono::system_clock::now();
+		const auto renderDt = std::chrono::duration_cast<std::chrono::milliseconds>(renderStart - lastTime).count();
+		lastTime = renderStart;
+		{ // Calculate Simple Moving Average for FrameTime
+			time_sum -= dtList[frame_count];
+			time_sum += renderDt;
+			dtList[frame_count] = renderDt;
+			frame_count++;
+			if (frame_count >= NBL_FRAMES_TO_AVERAGE)
+				frame_count = 0;
+		}
+		const double averageFrameTime = time_sum / (double)NBL_FRAMES_TO_AVERAGE;
```

Comment:
that average will be screwed up for the first `NBL_FRAMES_TO_AVERAGE-1` frames @Erfan-Ahmadi

Reply:
@AnastaZIuk See here for the fix: https://github.com/Devsh-Graphics-Programming/Nabla/commit/d41da30192ae26132457c807f4e5141592aa2fa0

Reply:
corrected